### PR TITLE
Add code to get content.

### PR DESCRIPTION
**Pull request by anghelutar**

### CODE REVIEW

File: alfresco-solrapi-client/alfresco-solrapi-client-spring/src/main/java/eu/xenit/alfresco/solrapi/client/spring/SolrApiSpringClient.java [hunk 1]
```diff
@@ -186,6 +193,77 @@ public SolrTransactions getTransactions(Long fromCommitTime, Long minTxnId, Long
...
+        try {
+            if(exchange.getBody()!=null)
+                content.setContent(exchange.getBody().getInputStream());
+            if(exchange.getHeaders().getContentType()!=null && exchange.getHeaders().getContentType().getCharset()!=null)
+                content.setContentEncoding(exchange.getHeaders().getContentType().getCharset().displayName());
+            List<String> transformStatus = exchange.getHeaders().get("X-Alfresco-transformStatus");
+            if(transformStatus!=null && !transformStatus.isEmpty())
+                content.setTransformStatusStr(transformStatus.get(0));
+            List<String> transformDuration = exchange.getHeaders().get("X-Alfresco-transformDuration");
+            if(transformDuration!=null )
```
Comment (driesdeproost) [inline alfresco-solrapi-client/alfresco-solrapi-client-spring/src/main/java/eu/xenit/alfresco/solrapi/client/spring/SolrApiSpringClient.java]: the formatting seems off in this file

File: alfresco-solrapi-client/alfresco-solrapi-client-spring/src/main/java/eu/xenit/alfresco/solrapi/client/spring/SolrApiSpringClient.java [hunk 2]
```diff
@@ -186,6 +193,77 @@ public SolrTransactions getTransactions(Long fromCommitTime, Long minTxnId, Long
...
+                content.setTransformDuration(transformDuration.get(0) != null ? Long.valueOf(transformDuration.get(0)) : null);
+            List<String> transformStatusException = exchange.getHeaders().get("X-Alfresco-transformException");
+            if(transformStatusException!=null && !transformStatusException.isEmpty())
+                content.setTransformException(transformStatusException.get(0));
+            content.setStatus(getStatus(exchange.getStatusCode(),transformStatus));
+        } catch (IOException e) {
+            e.printStackTrace();
```
Comment (driesdeproost) [inline alfresco-solrapi-client/alfresco-solrapi-client-spring/src/main/java/eu/xenit/alfresco/solrapi/client/spring/SolrApiSpringClient.java]: can you use the logger here?

File: alfresco-solrapi-client/alfresco-solrapi-client-spring/src/main/java/eu/xenit/alfresco/solrapi/client/spring/SolrApiSpringClient.java [hunk 3]
```diff
@@ -186,6 +193,77 @@ public SolrTransactions getTransactions(Long fromCommitTime, Long minTxnId, Long
...
+    @Override
+    public GetTextContentResponse getTextContent(Long nodeId, String propertyQName) {
+        UriComponentsBuilder uriBuilder = UriComponentsBuilder.fromHttpUrl(url).path("/textContent");
+        conditionalQueryParam(uriBuilder, "nodeId", nodeId, Objects::nonNull);
+        conditionalQueryParam(uriBuilder, "propertyQName", propertyQName, Objects::nonNull);
+
+        ResponseEntity<Resource> exchange = restTemplate
+                .exchange(uriBuilder.toUriString(), HttpMethod.GET, null, Resource.class);
+
+        GetTextContentResponse content = new GetTextContentResponse();
```
Comment (driesdeproost) [inline alfresco-solrapi-client/alfresco-solrapi-client-spring/src/main/java/eu/xenit/alfresco/solrapi/client/spring/SolrApiSpringClient.java:223]: this filling up of the content could be extracted in a separate function

File: alfresco-solrapi-client/alfresco-solrapi-client-spi/src/main/java/eu/xenit/alfresco/solrapi/client/spi/dto/GetTextContentResponse.java [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    public InputStream get() {
+        return content;
+    }
+
+    // no constructor, because it depends on the framework used
+
+    public static enum SolrApiContentStatus {
+        NOT_MODIFIED, OK, NO_TRANSFORM, NO_CONTENT, UNKNOWN, TRANSFORM_FAILED, GENERAL_FAILURE;
+
+        public static SolrApiContentStatus getStatus(String statusStr) {
```
Comment (driesdeproost) [inline alfresco-solrapi-client/alfresco-solrapi-client-spi/src/main/java/eu/xenit/alfresco/solrapi/client/spi/dto/GetTextContentResponse.java:24]: this is never used

Is this a remnant from before eu/xenit/alfresco/solrapi/client/spring/SolrApiSpringClient.java:228 ?

Comment (anghelutar) [reply alfresco-solrapi-client/alfresco-solrapi-client-spi/src/main/java/eu/xenit/alfresco/solrapi/client/spi/dto/GetTextContentResponse.java:24]: It is used, it's imported in SolrApiSpringClient.

File: alfresco-solrapi-client/alfresco-solrapi-client-spring/src/main/java/eu/xenit/alfresco/solrapi/client/spring/SolrApiSpringClient.java (hunk 3 above)
Comment (anghelutar) [reply alfresco-solrapi-client/alfresco-solrapi-client-spring/src/main/java/eu/xenit/alfresco/solrapi/client/spring/SolrApiSpringClient.java:223]: Ok, I did that.

File: alfresco-solrapi-client/alfresco-solrapi-client-spring/src/main/java/eu/xenit/alfresco/solrapi/client/spring/SolrApiSpringClient.java (hunk 1 above)
Comment (anghelutar) [reply alfresco-solrapi-client/alfresco-solrapi-client-spring/src/main/java/eu/xenit/alfresco/solrapi/client/spring/SolrApiSpringClient.java]: Not sure what the problem is here, intellij reports no indentation problem.

File: alfresco-solrapi-client/alfresco-solrapi-client-spring/src/main/java/eu/xenit/alfresco/solrapi/client/spring/SolrApiSpringClient.java (hunk 2 above)
Comment (anghelutar) [reply alfresco-solrapi-client/alfresco-solrapi-client-spring/src/main/java/eu/xenit/alfresco/solrapi/client/spring/SolrApiSpringClient.java]: Ok, I changed to the logger.

**Review by driesdeproost — APPROVED** [review]

**Pull request merged**